source/class/qxl/dialog/MForm.js

```diff
@@ -413,23 +615,76 @@ qx.Mixin.define("qxl.dialog.MForm", {
         /**
          * other widget properties @todo: allow to set all properties
          */
+
+        // width
         if (fieldData.width !== undefined) {
           formElement.setWidth(fieldData.width);
         }
-        if (fieldData.placeholder !== undefined) {
-          formElement.setPlaceholder(fieldData.placeholder);
+
+        // placeholder
+        if ( fieldData.placeholder !== undefined ) {
+          formElement.setPlaceholder( fieldData.placeholder );
+        }
+
+        // tooltip
+        if ( fieldData.toolTipText !== undefined ) {
+          formElement.setToolTipText( fieldData.toolTipText );
         }
+
+        // enabled
         if (fieldData.enabled !== undefined) {
           formElement.setEnabled(fieldData.enabled);
         }
 
+        // generic property setter
+        if ( typeof fieldData.properties == "object") {
+          formElement.set( fieldData.properties );
+        }
+
+        /*
+         * This allows changing the default autocomplete behavior to disable
+         * autocomplete on all text and password fields unless allowed at
+         * either the form level or at the field level using the
+         * allowBrowserAutocomplete key.
+         */
+        if ([
+              "textfield",
+              "passwordfield"
+            ].includes( fieldData.type.toLowerCase()) ) {
+
+          if ( typeof fieldData.allowBrowserAutocomplete == "boolean" ) {
+            if ( ! fieldData.allowBrowserAutocomplete ) {
+              //turn off autocomplete
+              formElement.getContentElement().setAttribute(
+              "autocomplete", "new-password");
+            } else {
+              // leave autocomplete alone.
+              // Note: Password field above sets attribute
+            }
+          } else if ( ! this.getAllowBrowserAutocomplete() ) {
+            //turn off autocomplete
+            formElement.getContentElement().setAttribute(
+              "autocomplete", "new-password");
+          }
+        }
+
+         // generic userdata settings
+        if ( typeof fieldData.userdata == "object" ) {
+          Object.keys( fieldData.userdata ).forEach(
+            function(key) {
+              formElement.setUserData(key, fieldData.userdata[key]);
+            });
+        }
+
         /**
          * Events
          */
         if (qx.lang.Type.isObject(fieldData.events)) {
           for (let type in fieldData.events) {
             try {
-              let func = eval("(" + fieldData.events[type] + ")"); // eval is evil, I know.
```

Comment:
This needs to stay how it is because `fieldData.events[type]` is a string (so that it can be transferred by json from the server). You can of course add a check for when it is a function to begin with.

Reply:
At least to stay BC. I don't know if I use that feature in my app, but to be safe, we shouldn't change it now. Maybe add a deprecation warning.

Reply:
BC in an incubator project, huh? :-)

Ok, I've re-added evil eval(string) a an option.

Reply:
I don't think the code is fit to be an incubator project that should be merged into core. It started as a quick & dirty solution with more and more ad-hoc features added over time, which is fine for a package. A "native" dialog should be rewritten from scratch.